### PR TITLE
freedv: 1.8.3.1 -> 1.8.4

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Supersedes #198804.

https://github.com/drowe67/freedv-gui/releases/tag/v1.8.4
https://github.com/drowe67/LPCNet/releases/tag/v0.3

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/lpcnetfreedv/default.nix [hunk 1]
```diff
@@ -1,30 +1,19 @@
 { lib, stdenv, fetchFromGitHub, fetchurl, cmake, codec2 }:
 
-let
```
Comment (illdefined) [inline pkgs/development/libraries/lpcnetfreedv/default.nix:3]: Without this, CMake attempts to fetch the data during the build on my machine (NixOS on x86-64) and fails.

File: pkgs/development/libraries/lpcnetfreedv/default.nix [hunk 2]
```diff
@@ -1,30 +1,19 @@
...
-  dataVersion = "191005_v1.0";
-  data = fetchurl {
-    url = "http://rowetel.com/downloads/deep/lpcnet_${dataVersion}.tgz";
-    sha256 = "1j1695hm2pg6ri611f9kr3spm4yxvpikws55z9zxizai8y94152h";
-  };
-in stdenv.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "lpcnetfreedv";
-  version = "unstable-2021-06-29";
+  version = "0.3";
```
Comment (illdefined) [inline pkgs/development/libraries/lpcnetfreedv/default.nix:5]: Using version `unstable-2022-08-22` / rev `67a6eb74d0c07faddcdce199856862cc45779d25` (current master) should solve the circular dependency issue (cf. <https://github.com/NixOS/nixpkgs/compare/master...illdefined:nixpkgs:freedv-1.8.4>)

File: pkgs/development/libraries/lpcnetfreedv/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/libraries/lpcnetfreedv/default.nix]: Reverted. Thanks!

File: pkgs/development/libraries/lpcnetfreedv/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/development/libraries/lpcnetfreedv/default.nix]: Done. Thanks!

**Comment by illdefined** [comment]
This looks good to me. I would have split up the commits, but that may be just a matter of taste.

I suggest that @markuskowa should have a quick glance at the changes to the `codec2` derivation. Now that the circular dependency has been resolved, it may or may not make sense to just enable LPCNet support (`-LPCNET:BOOL=ON`) unconditionally.

**Comment by illdefined** [comment]
Result of `nixpkgs-review pr 198852` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>freedv</li>
    <li>lpcnetfreedv</li>
  </ul>
</details>

**Comment by illdefined** [comment]
Successfully tested basic functionality of the `freedv` binary on NixOS (x86-64).

**Review by illdefined — APPROVED** [review]

**Review by markuskowa — APPROVED** [review]
LGTM

**Comment by illdefined** [comment]
@wegank, as you appear to be using FreeDV on Darwin, I have no objection to sharing maintenance responsibility. Feel free to add yourself to the list of maintainers at the next opportunity.

**Pull request merged**